script/app-coverage-report.js

```diff
@@ -27,34 +41,67 @@ const printCoverage = coverageResults => {
   console.log(coverageTable.toString());
 };
 
-const generateCoverage = (rootDir, coverageSummary) =>
-  Object.keys(coverageSummary).reduce((acc, coverageResult) => {
+const generateCoverage = (rootDir, coverageSummary) => {
+  // Get application manifests & entry names
+  const manifests = getAppManifests(path.join(__dirname, '../'));
+  const entryNames = manifests.map(manifest => manifest.entryName);
+
+  // Set initial coverage object
+  const initialCoverage = Object.assign(
+    ...entryNames.map(entryName => ({
+      [entryName]: {
+        path: manifests
+          .find(obj => obj.entryName === entryName)
+          .entryFile.replace(rootDir, ''),
+        lines: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        statements: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        functions: { total: 0, covered: 0, skipped: 0, pct: 0 },
+        branches: { total: 0, covered: 0, skipped: 0, pct: 0 },
+      },
+    })),
+  );
+
+  return Object.keys(coverageSummary).reduce((acc, coverageResult) => {
     const appName = coverageResult.replace(rootDir, '').split(path.sep)[0];
-    // If appName exists, add coverageResult to overall coverage
+
     if (appName) {
```

Comment:
Rather than nesting everything under an if, I think it might be better if a `.filter()` were used to remove any file that isn't under `src/applications/*`. I'm not sure if this should be done to the argument before it is passed or after.

Reply:
Yeah that makes sense. I think it would be better to call `.filter()` right before the `.reduce()` method is called. 

Technically we don't even need that initial check of whether the file is under `src/applications` because that's handled when storing the `coverageItem`. Only `coverageItems` with keys(filePaths) that are under `src/applications` will be updated. But filtering the results will prevent having to iterate through files that aren't under the applications folder.